### PR TITLE
add overview updates

### PR DESCRIPTION
**Pull request by tr0njavolta**

### CODE REVIEW

**Review by morsapaes — COMMENTED** [review]
Reviewing this in chunks: `isolation-level.md`

**Review by morsapaes — CHANGES_REQUESTED** [review]

**Comment by morsapaes** [comment]
In general, we should make sure this page aligns with the descriptions proposed by Jesse in the new get started guide. There's also some descriptions that can be lifted from the object-specific `CREATE` pages or the mentioned blogposts, to make sure we cover important details that are currently left out.

File: doc/user/content/overview/key-concepts.md [hunk 1]
```diff
@@ -10,45 +10,88 @@ aliases:
   - /overview/api-components/
 ---
 
-Materialize is a streaming database with a SQL API. However, despite the fact
-that Materialize uses SQL idioms and can process data from databases, it
-actually has very little in common with "databases" as most people think of them.
+Most physical and cloud-hosted database solutions require you to configure
```
Comment (jseldess) [inline doc/user/content/overview/key-concepts.md:13]: I think we need to more directly introduce the purpose of this page, and not focus on the benefits of a managed service in general. Suggestion (to start from):

```suggestion
This page introduces the key objects of Materialize.  
```

We might want to also say something like:

```
Some of these objects correspond traditional SQL idioms but have different properties and behaviors in the context of a streaming database like Materialize.
```

File: doc/user/content/overview/key-concepts.md [hunk 2]
```diff
@@ -58,7 +101,7 @@ Component      | Use
 
 #### Connectors
```
[the remark this answers is not in the document]
Comment (jseldess) [reply doc/user/content/overview/key-concepts.md:102]: I agree. If we take the "Getting data in" approach, we could intro with something like this (lifted and extended from my getting started draft):

> Materialize supports streaming data from multiple external sources, including [Kafka](/sql/create-source/kafka/) and [PostgreSQL](/sql/create-source/postgres/). The process for integrating a source involves working with three Materialize concepts: secrets, connections, and sources.

File: doc/user/content/overview/key-concepts.md [hunk 3]
```diff
@@ -80,148 +123,70 @@ Materialize can decode incoming bytes of data from several formats:
...
-
-Currently, Materialize only supports sending sink data to Kafka.
+## Data compute components
 
-## Views
+Component                | Use
+-------------------------|-----
+**Views**              | Queries of sources and other views that you want to save for repeated execution.
+**Indexes**            | Query results stored in memory.
+**Materialized views** | Query results stored durably.
```
[the remark this answers is not in the document]
Comment (jseldess) [reply doc/user/content/overview/key-concepts.md:141]: Agreed.

**Review by jseldess — CHANGES_REQUESTED** [review]
I agree with Marta's comments and left some of mine own. I appreciate all the time and effort put into this already, but given how much we lean on this page, we need to spend more time on this.

**Comment by tr0njavolta** [comment]
Closing pr draft to pull back on key-concepts doc as it needs more time to revise.

**Pull request closed without merging**